R/Lrnr_glm_semiparametric

```diff
@@ -0,0 +1,251 @@
+#' Semiparametric Generalized Linear Models
+#'
+#' This learner provides fitting procedures for semiparametric generalized linear models using a user-given baseline learner and
+#' \code{\link[stats]{glm.fit}}. It supports models of the form `linkfun(E[Y|A,W]) = linkfun(E[Y|A=0,W]) + A * f(W)` where `A` is a binary or continuous interaction variable,
+#' and `f(W)` is a user-specified parametric function (e.g. `f(W) = model.matrix(formula_sp, W)`). The baseline function `E[Y|A=0,W]` is fit using a user-specified Learner (possibly pooled over values of `A` and then projected onto the semiparametric model).
```

Comment:
I believe your package referenced a paper by Mark, Sherri, Susan for some of this methodology. If that's correct, please add this as a reference appropriately, including addition of the bibTeX entry into the `REFERENCES.bib` file in `inst/`; for how to cite, see, for example, https://github.com/tlverse/sl3/blob/devel/R/Lrnr_glmnet.R#L8.